### PR TITLE
Add tests and JavaDoc to classes of Task feature.

### PR DESCRIPTION
**Pull request by JoAndNight**
Description: This PR fixes the problems in implementation mentioned in PR#29. This PR adds unit tests and JavaDoc to all files of Task feature.

- [x]  Unit tests (For task feature): Before PR: 17% After PR: 50%
- [x]  Exception Handling: Exceptions are handled.
- [x]  JavaDocs Documentation
- [x]  SOLID principles
- [x]  Clean Architecture: Aligned with CA.

### CODE REVIEW

File: src/main/java/entities/TaskFactory.java [hunk 1]
```diff
@@ -59,4 +59,13 @@ public ArrayList<Task> convertToArray() {
         return taskList;
     }
 
+    /**
+     * Check if the given id is a key in the tasks HashMap of a TaskFactory.
```
Comment (EmilyDo) [inline src/main/java/entities/TaskFactory.java:63]: TaskFactory would need to change to TaskCollection

File: src/main/java/controllers/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package controllers;
+
```
Comment (MadhavKanna) [inline src/main/java/controllers/CategoryController.java:2]: Is it ok to add these files? the PR name mentions only changes to the Task feature. do resolve before merge. Same goes for the 4 category files here.

File: src/main/java/use_cases/tasks/create_task/CreateTask.java [hunk 3]
```diff
@@ -30,17 +35,14 @@ public CreateTask(CreateTaskOutputBoundary outputBoundary, TaskFactory taskFacto
...
-            String error = new String("Task Creation Failed. Please enter a valid deadline.");
+        if (inputData.getName().isBlank() | inputData.getDeadline().isLenient()){
+            String error = "Task Creation Failed. Please enter valid information";
             return outputBoundary.prepareFailView(error);
         }
 
         Task task = new Task(inputData.getName(), inputData.getDeadline());
         // The new Task object above is not assigned to any category.
-        // So we do not need to update the Category/CategoryCollection.
+        // So we do not need to update the Category/CategoryFactory.
```
Comment (MadhavKanna) [inline src/main/java/use_cases/tasks/create_task/CreateTask.java:45]: Do stick with previous naming convention of CategoryCollection

**Review by MadhavKanna — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/CategoryController.java (hunk 2 above)
Comment (JoAndNight) [reply src/main/java/controllers/CategoryController.java:2]: These are pulled from main, so i think it's fine

**Review by MadhavKanna — APPROVED** [review]

**Review by EmilyDo — APPROVED** [review]

**Pull request merged**